pytermgui/ansi_interface.py

```diff
@@ -107,243 +142,263 @@ class Color:
-        ):
+        if isinstance(color, str):
             try:
                 color = self.translate_hex(color)
             except ValueError:
                 # value is not a hex number, but is string
                 pass
 
         if color in self.names:
-            new = self.names[str(color)]
-            assert isinstance(new, int)
-            color = new
+            color = self.names.get(color, color)
 
-        # rgb values
-        if isinstance(color, tuple):
+        # 24-bit/RGB values (16581375 ≈16m colors)
+        if isinstance(color, tuple) and all([isinstance(part, int) for part in color]):
             red, green, blue = color
             color_value = f"2;{red};{green};{blue}m"
-
-        # 8-bit colors (including 0-16)
-        elif isinstance(color, int) or color.isdigit():
+        # 8-bit/Xterm colors (256 colors)
+        elif int(color) in range(256):
             color_value = f"5;{color}m"
-
         else:
             raise NotImplementedError(
                 f"Not sure what to do with {color} of type {type(color)}"
             )
+        ending = "\x1b[0m" if reset_style else ""
+        return f"\x1b[{38 + self.layer_offset};{color_value}{text}{ending}"
 
-        return (
-            f"\x1b[{38 + self.layer_offset};"
-            + color_value
-            + text
-            + (set_mode("reset") if reset_color else "")
-        )
-
-
-foreground = Color()
-"""`Color` instance to setting foreground colors"""
-
-background = Color(layer=1)
-"""`Color` instance to setting background colors"""
 
+foreground: Color = Color(0)
+"""
+`Color` instance to setting foreground colors
+"""
 
-def screen_size() -> tuple[int, int]:
-    """Get screen size using the os module
+background: Color = Color(1)
+"""
+`Color` instance to setting background colors
+"""
 
-    This is technically possible using a method of moving the
-    cursor to an impossible location, and using `report_cursor()`
-    to get where the position was clamped. This however messes
-    with the cursor position and printing gets a bit glitchy."""
 
-    try:
-        width, height = get_terminal_size()
-        return (width, height)
+class TerminalModes:
+    """
+    A class to change terminal modes
+    """
 
-    except OSError as error:
-        if error.errno in (9, 25):
-            return 0, 0
+    def __init__(self) -> None:
+        """
+        Initialize `TerminalModes` class
+        """
+        if _IS_NT:
+            self._OldStdinMode = ctypes.wintypes.DWORD()
+            self._OldStdoutMode = ctypes.wintypes.DWORD()
+            kernel32.GetConsoleMode(
+                kernel32.GetStdHandle(-10), ctypes.byref(self._OldStdinMode)
+            )
+            kernel32.GetConsoleMode(
+                kernel32.GetStdHandle(-11), ctypes.byref(self._OldStdoutMode)
+            )
+        else:
+            self._OldStdinMode = termios.tcgetattr(sys.stdin)
 
-        raise
+    @contextmanager
+    def NoEcho(self) -> None:
```

Comment:
I don't enjoy the PascalCase here. That is only supposed to be used by classes, which this is not. `no_echo` would work better.